LeaderboardBackend/Models/Entities/ApplicationContext.cs

```diff
@@ -17,6 +24,28 @@ public ApplicationContext(DbContextOptions<ApplicationContext> options)
     public DbSet<Run> Runs { get; set; } = null!;
     public DbSet<User> Users { get; set; } = null!;
 
+    public static NpgsqlDataSource CreateDataSource(PostgresConfig config)
```

Comment:
Move this class into AppContextDatsSourceProvider if it's only used there. Might even wanna make it an instance method or even delete it if it's only used in one place.